test/runtests.jl

```diff
@@ -195,12 +203,11 @@ end
         x,y = 0.1,0.2
         A, B, C = Fill(2,n), Zeros{Int}(n), Ones{Int}(n+1)
         
-
-        A_T, B_T, C_T = [1; fill(2,m-1)], fill(0,m), fill(1,m+1)
+        A_T, B_T, C_T = [1; fill(2,m-2)], fill(0,m-1), fill(1,m-1)
         @test clenshaw(vec(clenshaw(coeffs, x; dims=1)), A, B, C, y) ≈ clenshaw(vec(clenshaw(coeffs, A, B, C, y; dims=2)), x) ≈
                 only(clenshaw!([0.0], clenshaw!(Matrix{Float64}(undef,1,n), coeffs, x), A, B, C, y)) ≈ 
                 only(clenshaw!([0.0], clenshaw!(Matrix{Float64}(undef,m,1), coeffs, A, B, C, y), x)) ≈
-                forwardrecurrence(A_T, B_T, C_T, x)'coeffs*forwardrecurrence(A, B, C, y)
+                forwardrecurrence(A_T, B_T, C_T, x)'coeffs*forwardrecurrence(A, B, C, y)[1:end-1]
```

Comment:
I don't understand the `clenshaw` code enough to know if a similar change is needed there. In order to pass tests I am dropping the last element of `forwardrecurrence` to replicate the previous behavior.

From looking at `clenshaw` it seems that P0 isn't explicitly included, while it is in `forwardrecurrence`. Ideally, they would be consistent, but I'm not familiar with this domain enough to address it.

Reply:
Since `clenshaw` returns a single value there's no point in including `p0` as you can just multiply the returned value by `p0`